ocaml/src/proxy_server.ml

```diff
@@ -153,13 +153,34 @@ let read_objects_slices
                     ~fragment_statistics_cb
         >>= function
         | None -> Protocol.Error.failwith Protocol.Error.ObjectDoesNotExist
-        | Some (_mf, mf_src) -> Lwt.return mf_src
+        | Some (mf, namespace_id, mf_source) -> Lwt.return (object_name, mf, namespace_id, mf_source)
        )
        objects_slices
-     >>= fun mf_sources ->
+     >>= fun n_mf_src_s ->
+
+     let objects_infos =
```

Comment:
This whole transformation into object_infos is a bit weird (possibly correct but hard to follow).
I expected something more along the lines of
```
List.map
  (fun (name, mf, namespace_id, _) -> (name, "level0", (mf, namespace_id)))
  n_mf_src_s
```
?